runtime/src/main/kotlin/kotlin/text/regex/sets/CharSet.kt

```diff
@@ -28,13 +28,15 @@ package kotlin.text.regex
 open internal class CharSet(char: Char, val ignoreCase: Boolean = false) : LeafSet() {
 
     // We use only low case characters when working in case insensitive mode.
-    val char: Char = if (ignoreCase) char.toLowerCase() else char
+    @OptIn(ExperimentalStdlibApi::class)
+    val char: Char = if (ignoreCase) char.lowercaseChar() else char
```

Comment:
Changing usages are also not strictly needed for this PR.

Reply:
done